packages/lab/src/ProgressBar/ProgressBar.js

```diff
@@ -0,0 +1,125 @@
+import React from "react";
+import PropTypes from "prop-types";
+import clsx from "clsx";
+
+import clamp from "lodash/clamp";
+import { HvTypography } from "@hitachivantara/uikit-react-core";
+
+import { withStyles } from "@material-ui/core";
+
+import styles from "./styles";
+
+/**
+ * ProgressBar provides feedback about a process that is taking place in the application.
+ */
+const HvProgressBar = (props) => {
+  const { className, classes, value = 0, error, undeterminate = false, ...others } = props;
+
+  const clampedValue = clamp(value, 0, 100);
+
+  return (
+    <div
+      className={clsx(className, classes.root, classes.progress)}
+      role="progressbar"
+      aria-valuenow={clampedValue}
+      aria-valuemin="0"
+      aria-valuemax="100"
+      aria-busy={clampedValue !== 100}
+      {...(error && { "aria-invalid": true })}
+      {...others}
+    >
+      <div className={classes.progressContainer}>
+        {undeterminate ? (
+          <div />
+        ) : (
+          <div
+            style={{ width: `${clampedValue}%` }}
+            className={clsx(classes.progressBarLabel, clampedValue === 100 && classes.progressDone)}
+          >
+            <HvTypography variant="vizText">{`${clampedValue}%`}</HvTypography>
+          </div>
+        )}
+        <div className={classes.progressBarContainer}>
+          <div
+            style={{ width: `${clampedValue}%` }}
+            className={clsx(
+              classes.progressBar,
+              clampedValue === 100 && classes.progressDone,
+              error && classes.progressError
+            )}
+          />
+        </div>
+        <span
+          aria-live="polite"
+          style={{ display: "inline-block", width: "1px", height: "1px", overflow: "hidden" }}
+        >
+          {clampedValue}
+        </span>
```

Comment:
This logic could be fused with the label, and when it is in an undetermined state it hides the label but remains in the DOM